tests/src/test/scala/org/http4s/util/ProcessAppSpec.scala

```diff
@@ -0,0 +1,69 @@
+package org.http4s
+package util
+
+import scalaz.concurrent.Task
+import scalaz.stream.Process
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+
+class ProcessAppSpec extends Http4sSpec {
+
+  "ProcessApp" should {
+    "Terminate Server on a Process Failure" in {
+      val myError = new Throwable("Bad Initial Process")
+      class TestApp extends ProcessApp {
+        override def main(args: List[String]): Process[Task, Unit] = {
+          Process.fail(myError)
+        }
+      }
+      new TestApp().main(Array.empty[String]) should_== (())
+    }
+
+    "Terminate Server on a Valid Process" in {
+      class TestApp extends ProcessApp {
+        override def main(args: List[String]): Process[Task, Unit] = {
+          Process.empty[Task, String].append(Process.emit("Valid Process")).map(_ => ())
+        }
+      }
+      new TestApp().main(Array.empty[String]) should_== (())
+    }
+
+    "Terminate Server on a Bad Task" in {
+      val myError = new Throwable("Bad Task")
+      class TestApp extends ProcessApp {
+        override def main(args: List[String]): Process[Task, Unit] = {
+          Process.eval(Task.fail(myError))
+        }
+      }
+      new TestApp().main(Array.empty[String]) should_== (())
+    }
+
+    "Terminate Server on a Valid Task" in {
+      var touched = false
+      class TestApp extends ProcessApp {
+        override def main(args: List[String]): Process[Task, Unit] = {
+          Process.eval(Task{touched = true; ()})
+        }
+      }
+      new TestApp().main(Array.empty[String]) should_== (())
+    }
+
+    "requestShutdown Shuts Down a Server From A Separate Thread" in {
+      class TestApp extends ProcessApp {
+        override def main(args: List[String]): Process[Task, Unit] = {
+          Process.constant(())
```

Comment:
I've written this in a few places but not known what to call it so haven't stuck it in a util.  It runs forever and never emits.  It's what the server config `.serve` does in #1071.  Would be a good fit here, especially if we could name it:

```scala
Process.eval_(Task.async[Nothing] { _ => }))
```